spanner/database_leader_placement_samples.rb

```diff
@@ -0,0 +1,250 @@
+# Copyright 2021 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def instance_config project_id:, instance_config_id:
+  # [START spanner_get_instance_config]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_config_id = "Spanner instance config ID"
+
+  require "google/cloud/spanner"
+
+  instance_admin_client = \
+    Google::Cloud::Spanner::Admin::Instance::V1::InstanceAdmin::Client.new
+
+  instance_config_path = instance_admin_client.instance_config_path \
+    project: project_id, instance_config: instance_config_id
+  config = instance_admin_client.get_instance_config name: instance_config_path
+
+  puts "Available leader options for instance config #{config.name} : #{config.leader_options}"
+
+  # [END spanner_get_instance_config]
+end
+
+def list_instance_configs project_id:
+  # [START spanner_list_instance_configs]
+  # project_id  = "Your Google Cloud project ID"
+
+  require "google/cloud/spanner"
+
+  instance_admin_client = \
+    Google::Cloud::Spanner::Admin::Instance::V1::InstanceAdmin::Client.new
+
+  project_path = instance_admin_client.project_path project: project_id
+  configs = instance_admin_client.list_instance_configs parent: project_path
+
+  configs.each do |c|
+    puts "Available leader options for instance config #{c.name} : #{c.leader_options}"
+  end
+
+  # [END spanner_list_instance_configs]
+end
+
+def list_databases project_id:, instance_id:
+  # [START spanner_list_databases]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+
+  require "google/cloud/spanner"
+
+  db_admin_client = \
+    Google::Cloud::Spanner::Admin::Database::V1::DatabaseAdmin::Client.new
+
+  instance_path = db_admin_client.instance_path project: project_id,
+                                                instance: instance_id
+  databases = db_admin_client.list_databases parent: instance_path
+
+  databases.each do |db|
+    puts db.name
```

Comment:
We need to print out `default_leader`. Please note that if `default_leader` is not set explicitly, we will get an empty string.

Reply:
updated